### ISSUE TITLE
Proximal operators are not tested on ProductSpace

### ISSUE THREAD

**Issue by aringh** [odlgroup/odl#1533]
The bug in #1469 has not been caught in a test. This seems to be because proximal operator, as far as I can tell, are not systematically tested when the domain is `ProductSpace`.

- `proximal_operator_test` does not systematically test all of them, and the ones tested are not systematically tested on `ProductSpace`.
- `default_functionals_test` test the functionals on `ProductSpace`, but it does have extensive tests for all functionals.
- `functional_test` also have tests on `ProductSpace`, but the file only tests the derivative of the functionals and not their proximal operators.
- `default_functionals_slow_test` test all proximal operators of all default functionals. However, the tests are not performed on `ProductSpace`.

Adding a fixture for `ProductSpace` to the last of these test, i.e.,

```
    space = odl.uniform_discr(0, 1, 2)
    if product_space:
        space = odl.ProductSpace(space, 2)
```

in the `def functional` method, the result of the test is

```
7 failed, 316 passed, 414 skipped, 13 warnings, 160 error in 132.75 seconds
```

instead of 

```
202 passed, 247 skipped, 1 warnings in 74.06 seconds
```

**Comment by kohr-h** [comment]
Right, 7 failures is quite bad, and in particular that we don't have anything in the slow test suite is also not great.

On the other hand this

> in 132.75 seconds

versus this

> 74.06 seconds

shows why we chose not to test all cases everywhere in the regular test suite.

Anyway, the tests in [default_functionals_test.py](https://github.com/odlgroup/odl/blob/ed5dc2e044c100b10f24215d1b5c1a0a9e0be031/odl/test/solvers/functional/default_functionals_test.py) do cover product spaces, but are incomplete with respect to functionals tested. I'd suggest that we focus there to make the tests more comprehensive.
